Software/Development/SemesterProject/App/openLoopControlManager.c

```diff
@@ -60,74 +60,32 @@ int16 mapInt(int16 x, int16 xMax, int16 yMax)
 void setConstantDutyFromReference(void)
 {
     float duty;
-    float torqueRef, maxRef;
-    /*  INCLUDE THIS WHEN EVERYTHING IS ADDED
-    switch (getReferenceInput())
-    {
-        case POT:
-        {
-            torqueRef = getTorqueReferenceSliderMeasurement();
-            maxRef = getMaximumReferenceValue();
-        }break;
-        case GUI:
-        {
-            torqueRef = getTorqueReferenceFromGUI();
-            maxRef = getMaximumReferenceGUI();
-        }
-        case PEDAL:
-        {
-            torqueRef = getTorqueReferenceFromPedal();
-            maxRef = getMaximumReferencePedal();
-        }
-
-    }*/
-
-    torqueRef = getTorqueReferenceSliderMeasurement();
-    maxRef = 3.3;//getMaximumReferenceADC();
-
-    duty = mapFloat(torqueRef, maxRef, readMaxDuty());
+    float reference;
+
+    //Get the reference value to be set, this is a float between 0 and 100.
+    reference = getOpenLoopReference();
+
+    //Since reference and duty values have the same min and max value, there is no need for a mapping.
+    duty = reference;
```

Comment:
Do you think it's worth having the dummy variables?

I understand that you do it because previously this steps and conversions were necessary, however in the future if somebody looks at the code they would be like WTF. I think it's better like:

setAllDuties(getOpenLoopReference())

Not really looking at the use of 2 additional variables but for having a cleaner code. If you want I can add a description header to getOpenLoopReference() so it's clearer.

Reply:
Ok, will do